### PR TITLE
refactor: cleanup `Icon`

### LINKED ISSUE #7102: [Task]: Cleanup `Icon`

### PR DESCRIPTION
**Pull request by evavirseda**
## Summary

Closes #7102 

## Changelog

```
- Cleanup Icon component
- Create BoxedIcon component
- Add enums
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/IconBox.svelte [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<script lang="ts">
```
Comment (begonaalvarezd) [inline packages/shared/components/IconBox.svelte:1]: I would rename this component to BoxedIcon and make it so its a component that uses Icon but wrapps it in a box, so you could use it like `<BoxedIcon classes="text-blue-200" boxClasses="bg-pink-100" icon={IconEnum.Lola} />`

File: packages/shared/components/Alert.svelte [hunk 2]
```diff
@@ -2,36 +2,43 @@
...
     import { localize } from '@core/i18n'
     import { removeDisplayNotification } from '@auxiliary/notification/stores'
+    import { Icon as IconEnum } from '@auxiliary/icon'
 
     export let type: string
     export let message: string
     export let classes: string = ''
     export let showDismiss: boolean = false
     export let id: string = ''
+    interface ToastStyle {
```
Comment (begonaalvarezd) [inline packages/shared/components/Alert.svelte:12]: Lets add an empty space to separate the props for the interfaces 🙏🏼

File: packages/shared/components/inputs/Checkbox.svelte [hunk 3]
```diff
@@ -25,7 +25,13 @@
         }}
     >
         <Icon
-            icon={checked ? `checkbox${round ? '-round' : ''}` : `checkbox-unchecked${round ? '-round' : ''}`}
+            icon={checked
```
Comment (begonaalvarezd) [inline packages/shared/components/inputs/Checkbox.svelte:28]: I would move this logic to the script section and make if a bit easier to read with if elses, its a bit hard to read with nested ? ? ?

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
Amazing work, the code looks much better 💪🏼 
Added some comments to improve a bit more the code base 🙏🏼

**Review by cpl121 — CHANGES_REQUESTED** [review]
Nice job!
1 minimal comment

File: packages/shared/components/atoms/MenuItem.svelte [hunk 4]
```diff
@@ -1,5 +1,6 @@
 <script lang="ts">
     import { Icon, Text, Spinner, Position, InformationTooltip } from 'shared/components'
+    import { Icon as IconEnum } from '@auxiliary/icon'
 
     export let icon: string = ''
```
[the remark this answers is not in the document]
Comment (evavirseda) [reply packages/shared/components/atoms/MenuItem.svelte:5]: sure :pray:

**Review by cpl121 — CHANGES_REQUESTED** [review]
In addition to changing the `string` to `iconEnum`, I think it would be necessary to update the places where the `icon` was passed as in `string`
`CollectibleDetailsMenu` and `ExpirationTimePickerMenu`,

**Review by cpl121 — APPROVED** [review]
Perfect 🥇

**Pull request merged**